src/plugins/hiddenColumns/test/editors.e2e.js

```diff
@@ -36,4 +36,39 @@ describe('HiddenColumns', () => {
       expect($mainHolder.scrollLeft()).toBe(startScrollLeft);
     });
   });
+
+  describe('Data change by typing with hiddenColumns', () => {
+    it('should correctly render the changed values subjected to validation when there is a hidden column next to it', async() => {
+      const hot = handsontable({
+        data: [[1, 2, 'Smith']],
+        colHeaders: true,
+        hiddenColumns: {
+          columns: [0], // hide the first column
+          indicators: true,
+          copyPasteEnabled: false
+        },
+        columns: [
+          {}, // the first empty column
+          { // the second numeric column
+            type: 'numeric',
+            allowInvalid: false,
+          },
+          {}, // the last column without validation
+        ]
+      });
+
+      hot.selectCell(0, 1);
+      keyDown('enter'); // open the editor
+      await sleep(200);
+
+      document.activeElement.value = 'aa'; // type incorect value
+      keyDown('enter'); // confirm change the value
+      await sleep(200);
+      keyDown(Handsontable.helper.KEY_CODES.ESCAPE); // close the editor
```

Comment:
You can use `esc` instead.
`handsontableKeyTriggerFactory` will use proper keyCode internally: https://github.com/handsontable/handsontable/blob/dac8f5e68d8f21502183bb300e12395b36efa2be/test%2Fhelpers%2Fcommon.js#L390-L401